### PR TITLE
feat: Support generics on `messages` attribute in main `contract` call

### LINKED ISSUE #226: Allow implementing generic interface on contract
This should be done using `#[messages(my::module: query<Param: my + bonds>, exec<Msg: my + bonds>)]`.

### PR DESCRIPTION
**Pull request by jawoznia**
close #226

### CODE REVIEW

**Review by hashedone — COMMENTED** [review]
Seems mostly good, but I missed some tests. What happens if I try to implement the same interface twice with different generics? I think it should fail due to name collision. Also - can the contract itself be generic at this point? I believe not, right?

One thing I really miss is the tests of multitest utilities generated.

File: sylvia-derive/src/interfaces.rs [hunk 1]
```diff
@@ -104,28 +105,46 @@ impl Interfaces {
...
             .iter()
             .map(|interface| {
-                let enum_name = Self::merge_module_with_name(interface, msg_name);
-                let module = &interface.module;
-                quote! { &#module :: #enum_name :: messages()}
+                let ContractMessageAttr {
+                    module, generics, ..
+                } = interface;
+                let generics = if !generics.is_empty() {
+                    quote! { < #generics > }
```
[the remark this answers is not in the document]
Comment (jawoznia) [reply sylvia-derive/src/interfaces.rs:118]: Same

**Comment by jawoznia** [comment]
> Seems mostly good, but I missed some tests. What happens if I try to implement the same interface twice with different generics? I think it should fail due to name collision. Also - can the contract itself be generic at this point? I believe not, right?
> 
> One thing I really miss is the tests of multitest utilities generated.

I tested it with some more interfaces in this PR https://github.com/CosmWasm/sylvia/pull/239.
Is it cool to deliver it separately?

Yes you cannot implement same interface twice due to name collisions.

File: sylvia-derive/src/interfaces.rs [hunk 2]
```diff
@@ -90,11 +83,19 @@ impl Interfaces {
...
             .map(|interface| {
                 let ContractMessageAttr {
-                    module, variant, ..
+                    module,
+                    variant,
+                    generics,
+                    ..
                 } = interface;
+                let generics = if !generics.is_empty() {
+                    quote! { < #generics > }
```
[the remark this answers is not in the document]
Comment (hashedone) [reply sylvia-derive/src/interfaces.rs:92]: Good, my lack of knowledge here. But wanted to make sure you know what you are doing.

**Review by hashedone — APPROVED** [review]
With your explanations - LGTM

**Pull request merged**